src/os/macos.rs

```diff
@@ -42,12 +42,16 @@ impl Iterator for QueryIter {
 
     let mut depth = u32::MAX;
     let result = unsafe {
+      #[allow(clippy::ptr_as_ptr)]
+      let recurse_info =
+        (&mut info.protection as *mut _) as mach::vm_region::vm_region_recurse_info_t;
```

Comment:
I had to assign this to a variable because `#[allow(clippy::ptr_as_ptr)]` can't be applied to expressions.